### DISCUSSION TITLE
Error: inreplace failed

### DISCUSSION THREAD

**Discussion by mariano-daniel** [Homebrew/discussions#4586] [Everyday usage] [answered]
### Output of `brew config`

```
$ brew config
HOMEBREW_VERSION: 4.0.24
ORIGIN: https://github.com/Homebrew/brew
HEAD: 54c8876dc39047c04de15e7d212979ae8d98cf1c
Last commit: 2 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 8ba532b12931e94b61acc9058c45503e5a90a551
Core tap last commit: 42 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.OWROpyQsuM/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 4
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_INSTALL_FROM_API: set
Homebrew Ruby: 2.6.10 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: quad-core 64-bit skylake
Clang: 12.0.0 build 1200
Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 10.15.7-x86_64
CLT: 12.0.0.32.29
Xcode: N/A
```

### Output of `brew doctor`

```
$ brew doctor 
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You are using macOS 10.15.
We (and Apple) do not provide support for this old version.
It is expected behaviour that some formulae will fail to build in this old version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

### Description of issue

Trying to install openldap on Catalina, `Homebrew 4.0.24`

```
==> ./configure --prefix=/usr/local/Cellar/openldap/2.6.4 --sysconfdir=/usr/loca
==> make install
Error: inreplace failed
/usr/local/etc/openldap/slapd.conf:
  expected replacement of #<Pathname:/usr/local/Cellar/openldap/2.6.4> with #<Pathname:/usr/local/opt/openldap>
/usr/local/etc/openldap/slapd.ldif:
  expected replacement of #<Pathname:/usr/local/Cellar/openldap/2.6.4> with #<Pathname:/usr/local/opt/openldap>
```

Not sure how to proceed from here.

**Comment by fami-64** [comment, marked as answer]
I have the same problem, which I solved by manually modifying the 2 files.
In the "/usr/local/etc/openldap/slapd.conf" file, I changed line 17 to "# modulepath /usr/local/Cellar/openldap/2.6.4/libexec/openldap".
In the "/usr/local/etc/openldap/slapd.ldif" file, I changed line 32 to "#olcModulepath:	/usr/local/Cellar/openldap/2.6.4/libexec/openldap".
Then the installation/upgrade works

**Reply by gromgit** [reply]
Sounds like you had _existing_ OpenLDAP `slapd.*` config files, that the OpenLDAP source build (correctly) refused to overwrite, and therefore the formula's `inreplace` could not find the Cellar paths it expected to replace.

This is one of the arbitrary number of reasons that **Homebrew cannot officially support user builds**. Homebrew CI always builds bottles in "clean-slate" environments, while users' own machines almost always have excess stuff installed that inadvertently affects your builds (opportunistic linkages, failure to overwrite existing files with newer versions, etc.).

@mariano-daniel If you also have preexisting `/opt/homebrew/etc/slapd.*` config files, then you'll have to manually fix the paths as @fami-64 indicated, if you want your formula to build cleanly.

**Reply by mariano-daniel** [reply]
@fami-64 : You sir, are a genius! Thank you! I was under the very wrong impression that those files were not created and not present previously, so the install process was failing. Thanks for pointing me in the right direction so efficiently!

@gromgit : Thanks a lot for the detailed explanation! You both guys rock! If you have a Ko-fi account or PayPal or something let me know vía PM. I'd love to buy you a coffee or a beer as a token of appreciation, I really insist! I haven't run into many people like you that are super smart but also kind in order to help others. Thanks! :)

**Reply by qianbinbin** [reply]
@gromgit That makes sense, but some package managers like `apt` would let the user decide interactively, like keeping the old one, replacing with the new one, or opening a side by side diff view.

**Reply by gromgit** [reply]
@qianbinbin What you're talking about is installing from _prebuilt archives_ ("bottles" in Homebrew parlance), while the OP is talking about _local installation builds from source_. No package manager I know of supports incremental manual intervention if a command in a _build_ script fails, they just fail the entire build and expect you to sort it out from build logs, if any.

**Reply by qianbinbin** [reply]
@gromgit 

> What you're talking about is installing from prebuilt archives ("bottles" in Homebrew parlance), while the OP is talking about local installation builds from source.

I was actually building from source.

> No package manager I know of supports incremental manual intervention if a command in a build script fails, they just fail the entire build and expect you to sort it out from build logs, if any.

Got it. But users just expect `brew upgrade` to do similar things like before the support for the system was dropped.

**Reply by iwhyshop** [reply]
> I have the same problem, which I solved by manually modifying the 2 files. In the "/usr/local/etc/openldap/slapd.conf" file, I changed line 17 to "# modulepath /usr/local/Cellar/openldap/2.6.4/libexec/openldap". In the "/usr/local/etc/openldap/slapd.ldif" file, I changed line 32 to "#olcModulepath: /usr/local/Cellar/openldap/2.6.4/libexec/openldap". Then the installation/upgrade works

great job

**Comment by qianbinbin** [comment]
I encountered the same problem too, fixed by simply removing `/usr/local/etc/openldap`, for that it's just a dependence of `curl` to me.

**Reply by iwhyshop** [reply]
> I encountered the same problem too, fixed by simply removing `/usr/local/etc/openldap`, for that it's just a dependence of `curl` to me.

hope this works

**Comment by carlocab** [comment]
Homebrew/homebrew-core#135169 should make this problem go away in the future.
